### PR TITLE
fixing sklearn dep

### PR DESCRIPTION
**Pull request by ngam**
Follows #2057 and fixes tests: https://github.com/ngam/gpytorch/runs/7271650629?check_suite_focus=true

Recommendation: expand testing matrix to include python versions:

```yaml
    strategy:
      matrix:
        python-version: ["3.7", "3.8", "3.9", "3.10"]
    ...
    ...
      with:
        python-version: ${{ matrix.python-version }}
```

### CODE REVIEW

**Comment by gpleiss** [comment]
With this change in place, `python setup.py build develop` and `python setup.py install` now both fail on Python 3.7. I confirmed this on my machine.

I think this is okay, since installing gpytorch#master via pip (i.e. `pip install --upgrade git+https://github.com/cornellius-gp/gpytorch.git`) still works. This is what we recommend in the README.

@Balandat thoughts?

**Comment by adamjstewart** [comment]
`python setup.py *` is deprecated: https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html

**Comment by Balandat** [comment]
> @Balandat thoughts?

Aren’t we planning to remove py37 support in the next release (moving to LinearOperator - yes I still owe a review on that one)? In that case everything should be fine.

File: .github/workflows/run_test_suite.yml [hunk 1]
```diff
@@ -70,6 +70,7 @@ jobs:
     - name: Install dependencies
       run: |
         pip install pytest nbval jupyter tqdm matplotlib torchvision scipy
+        pip install -r requirements.txt
         python setup.py build develop
         pip install "pyro-ppl<=1.8";
```
[the remark this answers is not in the document]
Comment (gpleiss) [reply .github/workflows/run_test_suite.yml:75]: Yeah I think we can unpin.

[the remark this answers is not in the document]
Comment (Balandat) [reply .github/workflows/run_test_suite.yml:75]: Cool, we can do that in a separate PR I guess.

**Review by Balandat — APPROVED** [review]
gtmtl

**Pull request merged**